### PR TITLE
keep-sorted 0.5.0 (new formula)

### PR DESCRIPTION
**Pull request by tobinjt**
keep-sorted is a language-agnostic formatter that sorts lines between two markers in a larger file.  This is very useful for automatically sorting lists in code, and has support for blocks, nesting, line continuations, and more.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/k/keep-sorted.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  def install
+    system "go", "build", *std_go_args
+  end
+
+  test do
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # software. Run the test with `brew test keep-sorted`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+
```
Comment (chenrui333) [inline Formula/k/keep-sorted.rb:19]:
```suggestion
```

File: Formula/k/keep-sorted.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+class KeepSorted < Formula
+  desc "Language-agnostic formatter that sorts selectd lines"
+  homepage "https://github.com/google/keep-sorted"
+  url "https://github.com/google/keep-sorted/archive/refs/tags/v0.4.0.tar.gz"
+  sha256 "5f9ce09498ce8135749506a19f327332312d6aeb68cd4b30fe290a77b4ab1a4b"
```
Comment (chenrui333) [inline Formula/k/keep-sorted.rb:5]:
```suggestion
  url "https://github.com/google/keep-sorted/archive/refs/tags/v0.5.0.tar.gz"
  sha256 "8eee061af908fd971911118975e4a2870afff385b3aea9948cc9b221849a9436"
```

**Comment by tobinjt** [comment]
G'day,

Thanks for the super-fast review and helpful comments!

I have two questions:

- Given the version bump, should I withdraw this and open a new PR with a description saying `keep-sorted 0.5.0 (new formula)`?
- The `Commit style` check is failing, but I can't figure out why :(  I've followed https://docs.brew.sh/Formula-Cookbook#commit as far as I can see, and when I look at the tests tab I don't see it there.  Will you give me a pointer please to where I've made a mistake?

Thanks,

**Comment by carlocab** [comment]
> * The `Commit style` check is failing, but I can't figure out why :(  I've followed [docs.brew.sh/Formula-Cookbook#commit](https://docs.brew.sh/Formula-Cookbook#commit) as far as I can see, and when I look at the tests tab I don't see it there.  Will you give me a pointer please to where I've made a mistake?

The key points are:

> Please squash your commits into one with this message format, otherwise your PR will be replaced by our autosquash workflow.

> Now, [open a pull request](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) for your changes.
>
> - One formula per commit; one commit per formula.
> - Keep merge commits out of the pull request.

**Comment by tobinjt** [comment]
> > * The `Commit style` check is failing, but I can't figure out why :(  I've followed [docs.brew.sh/Formula-Cookbook#commit](https://docs.brew.sh/Formula-Cookbook#commit) as far as I can see, and when I look at the tests tab I don't see it there.  Will you give me a pointer please to where I've made a mistake?
> 
> The key points are:
> 
> > Please squash your commits into one with this message format, otherwise your PR will be replaced by our autosquash workflow.
> 
> > Now, [open a pull request](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) for your changes.
> > 
> > * One formula per commit; one commit per formula.
> > * Keep merge commits out of the pull request.

Hey Carlo,

Thanks for the info, I think I have successfully done that.

Thanks,

File: Formula/k/keep-sorted.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  desc "Language-agnostic formatter that sorts selectd lines"
+  homepage "https://github.com/google/keep-sorted"
+  url "https://github.com/google/keep-sorted/archive/refs/tags/v0.5.0.tar.gz"
+  sha256 "8eee061af908fd971911118975e4a2870afff385b3aea9948cc9b221849a9436"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args
```
Comment (chenrui333) [inline Formula/k/keep-sorted.rb:11]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w")
```

**Review by chenrui333 — APPROVED** [review]
just one more change

File: Formula/k/keep-sorted.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+class KeepSorted < Formula
+  desc "Language-agnostic formatter that sorts selectd lines"
```
Comment (carlocab) [inline Formula/k/keep-sorted.rb:2]:
```suggestion
  desc "Language-agnostic formatter that sorts selected lines"
```

**Comment by tobinjt** [comment]
Thanks Carlo and Rui!

I applied your changes, squashed the commits, and pushed.  Now I need another approval please?

Thanks,

**Review by chenrui333 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Comment by thiagowfx** [comment]
```
% keep-sorted --version
unknown
```

I'll file a new issue, but wanted to note it here.

**Pull request merged**